types/src/v0/impls/state.rs

```diff
@@ -316,15 +316,16 @@ pub fn validate_proposal(
         });
     }
 
-    // Check if timestamp is increasing (with some tolerance).
-    if proposal.timestamp() <= parent_header.timestamp() - 2 {
+    // Check if timestamp is increasing.
+    let system_time: u64 = OffsetDateTime::now_utc().unix_timestamp() as u64;
+    if proposal.timestamp() < system_time - 2 {
```

Comment:
We _also_ want to enforce the timestamp is increasing relative to the parent. Ie the condition for validity should be something like `proposal.timestamp() >= parent_header.timestamp() && proposal.timestamp().abs_diff(system_time) <= 2`

Reply:
dc197d664cf7cc2368500b08ed584a52b4c9398f